nilearn/_utils/testing.py

```diff
@@ -56,6 +60,34 @@ def assert_warns(warning_class, func, *args, **kw):
         return output
 
 
+def assert_warns_regex(warning_class, warning_regex,
```

Comment:
I may have missed it but it looks like you are not using this function in this PR (kind of makes sense that it belongs in the warnings-related PR)